virttest/utils_misc.py

```diff
@@ -3331,34 +3331,35 @@ class struct_timespec(ctypes.Structure):
         return monotonic_time_os()
 
 
-def verify_host_dmesg(dmesg_log_file=None, trace_re=None):
+def verify_host_dmesg(dmesg_log_file=None, ignore_result=False, level_check=3):
     """
     Find host call trace in dmesg log.
 
     :param dmesg_log_file: The file used to save host dmesg. If None, will save
                            host dmesg to logging.debug.
-    :param trace_re: re string used to filter call trace.
-    """
-    panic_re = [r"BUG:.*---\[ end trace .* \]---"]
-    panic_re.append(r"----------\[ cut here.* BUG .*\[ end trace .* \]---")
-    panic_re.append(r"general protection fault:.* RSP.*>")
-    panic_re.append(r"Out of memory: Kill process .*Call Trace:")
-    panic_re = "|".join(panic_re)
-    dmesg = process.system_output("dmesg", timeout=30, ignore_status=True,
-                                  verbose=False)
-    if dmesg:
-        match = re.search(panic_re, dmesg, re.DOTALL | re.MULTILINE | re.I)
-        if match is not None:
-            err = "Found call trace in host dmesg."
-            d_log = "Host dmesg log:\n%s" % dmesg
-            if dmesg_log_file:
-                file(dmesg_log_file, "w+").write(d_log)
-                err += " Please check host dmesg log %s." % dmesg_log_file
-            else:
-                err += " Please check host dmesg log in debug log."
-                logging.debug(d_log)
-            process.system("dmesg -C", ignore_status=True)
-            raise exceptions.TestError(err)
+    :param ignore_result: True or False, whether to fail test case on issues
+    :param level_check: level of severity of issues to be checked
+                        1 - emerg
+                        2 - emerg,alert
+                        3 - emerg,alert,crit
+                        4 - emerg,alert,crit,err
+                        5 - emerg,alert,crit,err,warn
+    """
+    cmd = "dmesg -T -l %s|grep ." % ",".join(map(str, range(1, int(level_check) + 1)))
```

Comment:
Btw not sure what the default should be, but on my laptop I get:

```
$ dmesg -T -l 2
[Pá čec  7 00:14:57 2017] CPU1: Core temperature above threshold, cpu clock throttled (total events = 1)
[Pá čec  7 00:14:57 2017] CPU0: Core temperature above threshold, cpu clock throttled (total events = 1)
[Pá čec  7 00:14:57 2017] CPU4: Package temperature above threshold, cpu clock throttled (total events = 1)
[Pá čec  7 00:14:57 2017] CPU5: Package temperature above threshold, cpu clock throttled (total events = 1)
[Pá čec  7 00:14:57 2017] CPU6: Package temperature above threshold, cpu clock throttled (total events = 1)
[Pá čec  7 00:14:57 2017] CPU7: Package temperature above threshold, cpu clock throttled (total events = 1)
[Pá čec  7 00:14:57 2017] CPU3: Package temperature above threshold, cpu clock throttled (total events = 1)
...
```

and

```
dmesg -T -l 3
[Ne čec  2 21:29:15 2017] mmc0: Unknown controller version (3). You may experience problems.
[Ne čec  2 21:29:27 2017] tpm tpm0: A TPM error (6) occurred attempting to read a pcr value
[Ne čec  2 21:29:35 2017] Bluetooth: hci0: turning off Intel device LED failed (-19)
[Ne čec  2 22:55:39 2017] iwlwifi 0000:03:00.0: Failed to find station
[Ne čec  2 22:55:39 2017] wlp3s0: failed to remove key (1, ff:ff:ff:ff:ff:ff) from hardware (-22)
[Ne čec  2 22:55:44 2017] tpm tpm0: A TPM error (6) occurred attempting to read a pcr value
[Po čec  3 01:40:19 2017] iwlwifi 0000:03:00.0: Failed to find station
[Po čec  3 01:40:19 2017] wlp3s0: failed to remove key (1, ff:ff:ff:ff:ff:ff) from hardware (-22)
[Po čec  3 01:40:23 2017] tpm tpm0: A TPM error (6) occurred attempting to read a pcr value
[Po čec  3 01:41:08 2017] iwlwifi 0000:03:00.0: Failed to find station
[Po čec  3 01:41:08 2017] wlp3s0: failed to remove key (1, ff:ff:ff:ff:ff:ff) from hardware (-22)
[Po čec  3 01:41:12 2017] tpm tpm0: A TPM error (6) occurred attempting to read a pcr value
[Po čec  3 02:36:05 2017] usb 3-3: device not accepting address 22, error -71
[Po čec  3 05:53:03 2017] usb 3-3: device not accepting address 31, error -71
[Po čec  3 05:53:23 2017] usb 3-3: device not accepting address 62, error -71
[Po čec  3 05:53:24 2017] usb 3-3: device not accepting address 64, error -71
[Po čec  3 05:53:56 2017] usb 3-3: device not accepting address 107, error -71
[Po čec  3 05:55:42 2017] usb 3-3: device not accepting address 17, error -71
[Po čec  3 05:56:18 2017] usb 3-3: device not accepting address 43, error -71
[Po čec  3 14:38:25 2017] iwlwifi 0000:03:00.0: Failed to find station
[Po čec  3 14:38:25 2017] wlp3s0: failed to remove key (1, ff:ff:ff:ff:ff:ff) from hardware (-22)
[Po čec  3 14:38:29 2017] tpm tpm0: A TPM error (6) occurred attempting to read a pcr value
[Út čec  4 13:30:55 2017] ata6.00: exception Emask 0x0 SAct 0x0 SErr 0x0 action 0x6
[Út čec  4 13:30:55 2017] ata6.00: waking up from sleep
[Út čec  4 13:34:58 2017] iwlwifi 0000:03:00.0: Failed to find station
[Út čec  4 13:34:58 2017] wlp3s0: failed to remove key (1, ff:ff:ff:ff:ff:ff) from hardware (-22)
[Út čec  4 13:35:02 2017] tpm tpm0: A TPM error (6) occurred attempting to read a pcr value
[Út čec  4 13:35:03 2017] ata6: exception Emask 0x10 SAct 0x0 SErr 0x0 action 0xe frozen
[Út čec  4 13:35:03 2017] ata6: ACPI event
[Út čec  4 13:36:36 2017] iwlwifi 0000:03:00.0: Failed to find station
[Út čec  4 13:36:36 2017] wlp3s0: failed to remove key (1, ff:ff:ff:ff:ff:ff) from hardware (-22)
[Út čec  4 13:36:39 2017] tpm tpm0: A TPM error (6) occurred attempting to read a pcr value
[Út čec  4 13:36:44 2017] ata6.00: exception Emask 0x0 SAct 0x0 SErr 0x0 action 0x6
[Út čec  4 13:36:44 2017] ata6.00: waking up from sleep
[Út čec  4 13:38:56 2017] sd 6:0:0:0: [sdb] No Caching mode page found
[Út čec  4 13:38:56 2017] sd 6:0:0:0: [sdb] Assuming drive cache: write through
[Út čec  4 14:04:22 2017] iwlwifi 0000:03:00.0: Failed to find station
[Út čec  4 14:04:22 2017] wlp3s0: failed to remove key (1, ff:ff:ff:ff:ff:ff) from hardware (-22)
[Út čec  4 14:04:27 2017] tpm tpm0: A TPM error (6) occurred attempting to read a pcr value
[Út čec  4 14:04:27 2017] ata6: exception Emask 0x10 SAct 0x0 SErr 0x0 action 0xe frozen
[Út čec  4 14:04:27 2017] ata6: ACPI event
[Út čec  4 14:24:20 2017] iwlwifi 0000:03:00.0: Failed to find station
...
```

Which means I'd get frequent test failures even just because slightly overheated host or because unsupported mmc controller. I think we need to check how frequent this is on other systems...